### PR TITLE
Add missing comments

### PR DESCRIPTION
**Pull request by philbrookes**
# Motivation
Switched IDE to Visual Studio and it alerted me to many missing comments I had left.

# Changes
Passed through and added them all in now.

### CODE REVIEW

File: analysis.go [hunk 1]
```diff
@@ -22,6 +24,7 @@ type Result struct {
 	Events        []Event `json:"events" yaml:"events"`
 }
 
+// Event Is a representation of the items in the openshift event log, this is trimmed to only the required fields
```
Comment (rhcarvalho) [inline analysis.go]: s/Is/is

File: analysis.go [hunk 2]
```diff
@@ -34,10 +37,12 @@ type Event struct {
 	Type    string `json:"type"`
 }
 
+// Events is a representation of everything in the openshift event log for a particular project
 type Events struct {
 	Items []Event `json:"items"`
 }
 
+// DeploymentConfigs Is a representation of the openshift DeploymentConfigs JSON
```
Comment (rhcarvalho) [inline analysis.go]: s/Is/is

Comment (rhcarvalho) [inline analysis.go]: OpenShift

File: analysis.go [hunk 3]
```diff
@@ -51,11 +56,13 @@ type DeploymentConfigs struct {
 	} `json:"items"`
 }
 
+// ContainerWaiting is a possible status that a pods containers can be in.
```
Comment (rhcarvalho) [inline analysis.go]: Just a question, are we reinventing this or taking it from origin?

File: analysis.go [hunk 4]
```diff
@@ -51,11 +56,13 @@ type DeploymentConfigs struct {
...
 type ContainerWaiting struct {
 	Reason  string `json:"reason"`
 	Message string `json:"message"`
 }
 
+// Pods is a representation of all the pods in a project from openshift.
```
Comment (rhcarvalho) [inline analysis.go]: OpenShift

File: tasks.go [hunk 5]
```diff
@@ -215,6 +216,10 @@ func RunAllAnalysisTasks(runner Runner, path string, workers int) {
 	log.Printf("Run %d analysis tasks in %v.", taskCount, delta)
 }
 
+// GetAllAnalysisTasks returns a channel of all tasks known to the dump tool. It returns
+// immediately and sends tasks to the channel in a separate goroutine. The
+// channel is closed after all tasks are sent.
+// FIXME: GetAllAnalysisTasks should not need to know about basepath.
```
Comment (rhcarvalho) [inline tasks.go]: There is no "basepath" in the signature.

**Comment by rhcarvalho** [comment]
I would suggest consistently ending the comments with a ".".

**Comment by philbrookes** [comment]
@rhcarvalho does this look OK to merge now?

File: analysis.go [hunk 6]
```diff
@@ -22,6 +26,8 @@ type Result struct {
 	Events        []Event `json:"events" yaml:"events"`
 }
 
+// Event is a representation of the items in the openshift event log, this is
```
Comment (rhcarvalho) [inline analysis.go]: OpenShift

File: analysis.go [hunk 7]
```diff
@@ -34,10 +40,13 @@ type Event struct {
 	Type    string `json:"type"`
 }
 
+// Events is a representation of everything in the openshift event log for a
```
Comment (rhcarvalho) [inline analysis.go]: OpenShift

File: analysis.go [hunk 8]
```diff
@@ -34,10 +40,13 @@ type Event struct {
...
+// particular project.
 type Events struct {
 	Items []Event `json:"items"`
 }
 
+// DeploymentConfigs is a representation of the OpenShift DeploymentConfigs JSON.
```
Comment (rhcarvalho) [inline analysis.go]: "DeploymentConfigs represents OpenShift deployment configs." ?

(it doesn't have to be JSON...)

File: analysis.go [hunk 9]
```diff
@@ -51,11 +60,15 @@ type DeploymentConfigs struct {
 	} `json:"items"`
 }
 
-type ContainerWaiting struct {
+// ContainerStateWaiting is a possible status that a pods containers can be in.
+// This structure is cloned from:
+// https://github.com/openshift/origin/blob/570a4a3d0eb4c56d1b7cb1eef31ce9adcc0ec95a/vendor/k8s.io/kubernetes/pkg/api/types.go#L1052
```
Comment (rhcarvalho) [inline analysis.go]: Hmm, I know I suggested this :)

But I meant it more generally... why would we add a note with such a link to one of the types and leave all the others without?

We could tell where they come from in a common comment preceding all definitions, and the other suggestion was to copy doc comment + type definition from origin, then remove the struct fields we don't care about.

File: millicore.go [hunk 10]
```diff
@@ -5,6 +5,10 @@ import (
 	"path/filepath"
 )
 
+// GetMillicoreConfigTasks will create a function matching the CheckTask
+// interface for each project to retrieve the millicore config in that
```
Comment (rhcarvalho) [inline millicore.go]: not sure: is it millicore or Millicore?

File: tasks.go [hunk 11]
```diff
@@ -215,7 +216,11 @@ func RunAllAnalysisTasks(runner Runner, path string, workers int) {
 	log.Printf("Run %d analysis tasks in %v.", taskCount, delta)
 }
 
-func GetAllAnalysisTasks(runner Runner, path string, results chan<- CheckResults) <-chan Task {
+// GetAllAnalysisTasks returns a channel of all tasks known to the dump tool. It returns
```
Comment (rhcarvalho) [inline tasks.go]: AFAIU it doesn't return "_all tasks_ known to the dump tool", I think tasks were split between dump and analysis?

File: millicore.go (hunk 10 above)
Comment (philbrookes) [inline millicore.go]: I'm not sure either, but looking around, it seems to be referred to as Millicore, so I will change here.

**Comment by rhcarvalho** [comment]
LGTM

**Pull request merged**